src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  let currentState = { ...state };
```

Comment:
Rename the object copy to `currentStateCopy` instead of `currentState` to better reflect its purpose and to follow the naming conventions outlined in the checklist.